support/util/util.go

```diff
@@ -141,3 +143,23 @@ func decompress(r io.Reader) (*bytes.Buffer, error) {
 
 	return bytes.NewBuffer(data), nil
 }
+
+// IsUrlReachable receives a url string and checks if the url is resolvable and reachable
+// Port 0 is used by default if no port is specfied in the url
+func IsUrlReachable(url string) bool {
+	host := url
+	port := ""
+	if strings.Contains(url, ":") {
+		address := strings.Split(url, ":")
+		host = address[0]
+		port = address[1]
+	}
+
+	timeout := 1 * time.Second
+	conn, err := net.DialTimeout("udp", fmt.Sprintf("%s:%s", host, port), timeout)
```

Comment:
Why `udp`?

Reply:
`icmp` requires privileged access,
`tcp` we need to know the exact opened port we want to connect to, and the host has to also acknowledge the connection.

Reply:
can we do https://github.com/openshift/hypershift/pull/2059/files#r1081142457 instead?

Reply:
working on it